src/module/token/transferable/CreatorTokenERC20.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache 2.0
+pragma solidity ^0.8.0;
+
+import {ModularModule} from "../../../ModularModule.sol";
+import {Role} from "../../../Role.sol";
+import {BeforeTransferCallbackERC20} from "../../../callback/BeforeTransferCallbackERC20.sol";
+
+import {ICreatorToken} from "@limitbreak/creator-token-standards/interfaces/ICreatorToken.sol";
+
+import {ITransferValidator} from "@limitbreak/creator-token-standards/interfaces/ITransferValidator.sol";
+import {ITransferValidatorSetTokenType} from
+    "@limitbreak/creator-token-standards/interfaces/ITransferValidatorSetTokenType.sol";
+import {TOKEN_TYPE_ERC20} from "@limitbreak/permit-c/Constants.sol";
+
+library CreatorTokenStorage {
+
+    /// @custom:storage-location erc7201:creator.token.erc20
+    bytes32 public constant CREATORTOKEN_STORAGE_POSITION =
+        keccak256(abi.encode(uint256(keccak256("creator.token.erc20")) - 1)) & ~bytes32(uint256(0xff));
+
+    struct Data {
+        // the addresss of the transfer validator
+        address transferValidator;
+    }
+
+    function data() internal pure returns (Data storage data_) {
+        bytes32 position = CREATORTOKEN_STORAGE_POSITION;
+        assembly {
+            data_.slot := position
+        }
+    }
+
+}
+
+contract CreatorTokenERC20 is ModularModule, BeforeTransferCallbackERC20, ICreatorToken {
+
+    /*//////////////////////////////////////////////////////////////
+                                ERRORS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Revert with an error if the transfer validator is not valid
+    error InvalidTransferValidatorContract();
+
+    /*//////////////////////////////////////////////////////////////
+                            EXTENSION CONFIG
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Returns all implemented callback and extension functions.
+    function getModuleConfig() external pure override returns (ModuleConfig memory config) {
+        config.callbackFunctions = new CallbackFunction[](1);
+        config.fallbackFunctions = new FallbackFunction[](3);
+
+        config.callbackFunctions[0] = CallbackFunction(this.beforeTransferERC20.selector);
+
+        config.fallbackFunctions[0] =
+            FallbackFunction({selector: this.getTransferValidator.selector, permissionBits: 0});
+        config.fallbackFunctions[1] =
+            FallbackFunction({selector: this.getTransferValidationFunction.selector, permissionBits: 0});
+        config.fallbackFunctions[2] =
+            FallbackFunction({selector: this.setTransferValidator.selector, permissionBits: Role._MANAGER_ROLE});
+        config.requiredInterfaces = new bytes4[](1);
+        config.requiredInterfaces[0] = 0x36372b07; // ERC20
```

Comment:
Need to double check ERC20Core because the actual ERC20 spec doesn't requires ERC165